File: Backend/Database/Models/Feedback.cs
using System.ComponentModel.DataAnnotations;

namespace Database.Models;

public class Feedback
{
    [Key]
    public required Guid Id { get; set; }
    public required string Comment { get; set; }
    public required Guid AssignmentId { get; set; }
    public Assignment? Assignment { get; set; }
    public required Guid? StudentId { get; set; }
    public User? Student { get; set; }
    public required Guid? TeamId { get; set; }
    public Team? Team { get; set; }
}

public class ApprovalFeedback : Feedback
{
    public required bool IsApproved { get; set; }
}

public enum LetterGrade
{
    A, B, C, D, E, F
}

public class LetterFeedback : Feedback
{
    public required LetterGrade LetterGrade { get; set; }
}

public class PointsFeedback : Feedback
{
    public required int Points { get; set; }
}

File: Frontend/src/types/feedback.ts
export type LetterGrade = 'A' | 'B' | 'C' | 'D' | 'E' | ' F';

export type Feedback = {
    id: string;
    comment: string;
    assignmentId: string;
    studentId?: string;
    teamId?: string;
    isApproved?: boolean;
    letterGrade?: LetterGrade;
    points?: number;
}

export type CreateFeedback = {
    comment: string;
    assignmentId: string;
    studentId?: string;
    teamId?: string;
    isApproved?: boolean;
    letterGrade?: LetterGrade;
    points?: number;
}

export type EditFeedback = {
    comment: string;
    isApproved?: boolean;
    letterGrade?: LetterGrade;
    points?: number;
}